### ISSUE TITLE
Should not allow multiple Subprotocols for WebSocket Client Handshake

### ISSUE THREAD

**Issue by irunika** [netty/netty#7166]
According to the specification of WebSocket sub-protocols in [RFC 6455 section 1.9](https://tools.ietf.org/html/rfc6455#section-1.9) even though the server could have multiple subprotocols which can be allowed, a client can only ask for a specific subprotocol in the handshake. But in Netty even the client comes with multiple subprotocols, the server accepts it. IMHO this is not the way which the handshake should be done with subprotocols.

**Comment by normanmaurer** [comment]
@irunika would you be interested in providing a PR for a fix with a unit test ?

**Comment by irunika** [comment]
@normanmaurer I'm busy with my work until next week and I'm using WebSocket and in the testing, I found this issue. If you guys are not in a hurry. I will send a PR ASAP.

**Comment by irunika** [comment]
K sure. I will send a PR.

**Comment by normanmaurer** [comment]
@irunika thanks!

**Comment by irunika** [comment]
@normanmaurer I looked into your code. I think I was wrong about the behavior. I checked with Wireshark and found out that netty is responding with the first found compatible sub protocol. Not all. So that is fine. <br> 
But even though the supported sub protocols are null in the server and if client comes asking some sub protocol handshake is happening without **sec-websocket-protocol** header in the response which means that it does not support sub protocols. Is this a good behavior? Because even the client is asking for sub protocols netty handshakes and say it does not support them by not sending the **sec-websocket-protocol** header.
